### DISCUSSION TITLE
Get Parameter variable and use it in different tasks

### DISCUSSION THREAD

**Discussion by matlh0** [PrefectHQ/prefect#5127] [Q&A]
Hi, 

I need to get Parameter en put them in some tasks.

i've test this : 

```
import prefect
from prefect import task, Flow, Parameter

logger = prefect.context.get("logger")

@task
def step_1(arg):
    logger.info(f"Step 1 : arg : {arg}!")

@task
def step_2(arg):
    logger.info(f"Step 2 : arg : {arg}!")

with Flow("get_params") as flow:
    hostname = Parameter('hostname', default = "None")
    task_1 = step_1(hostname)
    task_2 = step_2(arg=hostname)
    task_1.set_dependencies(downstream_tasks=[task_2])
    task_2.set_dependencies(upstream_tasks=[task_1])

flow.run()
```

But the schematic is not good 
![Capture d’écran 2021-11-10 à 08 32 04](https://user-images.githubusercontent.com/90396170/141069332-7db84dd7-eb35-4dea-8d9d-c4b052faeeb9.png)

I'm trying to do this : 
![Capture d’écran 2021-11-10 à 08 34 09](https://user-images.githubusercontent.com/90396170/141069726-1e238273-afb7-4541-bb01-6397ab98a271.png)

but I can't get the parameters and pass them as arguments in the spots 

Thanks for your help :) 

Matt

**Comment by anna-geller** [comment]
@matlh0 to clarify: you want to ensure that step_2 runs after step_1 task, and that both of those tasks receive the Parameter value of the Parameter hostname? if so, the way you did it is perfectly fine, even though you don't need to set the dependency between step_1 and step_2 twice, only either downstream or upstream should suffice:

```python
with Flow("get_params") as flow:
    hostname = Parameter('hostname', default = "None")
    task_1 = step_1(hostname)
    task_2 = step_2(hostname)
    task_1.set_dependencies(downstream_tasks=[task_2])
```
![image](https://user-images.githubusercontent.com/86264395/141151737-24b62633-da79-4da7-9aa6-90af048976f5.png)

But if you want that only step_1 receives the parameter, then you can use this:

```python
with Flow("get_params") as flow:
    hostname = Parameter('hostname', default = "None")
    task_1 = step_1(hostname)
    task_2 = step_2()
    task_1.set_dependencies(downstream_tasks=[task_2])
```
![image](https://user-images.githubusercontent.com/86264395/141152005-d6ab3114-5b40-4c22-bd46-3b53e701436f.png)
